lib/SILOptimizer/Mandatory/DIMemoryUseCollector.cpp

```diff
@@ -835,6 +850,11 @@ void ElementUseCollector::collectUses(SILValue Pointer, unsigned BaseEltNo) {
       continue;
     }
 
+    // Sanitizer instrumentation is not user visible, so it should not
+    // count as a use and must not affect compile-time diagnostics.
+    if (isSanitizerInstrumentation(User, Module.getASTContext()))
+      continue;
```

Comment:
We can probably remove any `tsanInoutAccess` operations that happen prior to a storage slot's definite initialization, since the DI analysis should prevent escaping and ensure races are impossible before the initialization occurs.